docs/optimizer/optimize.md

```diff
@@ -0,0 +1,52 @@
+# Optimizing a Model using the Optimizer
+
+## Introduction
+
+The ONNX Script `Optimizer` tool provides the user with the functionality to optimize an ONNX model by performing optimizations and clean-ups such as constant folding, dead code elimination, etc.
+
+## Usage
+
+In order to utilize the optimizer tool,
+
+```python
+    import onnxscript
+
+    onnxscript.optimizer.optimize(model)
+```
+
+### optimize API
+The `onnxscript.optimizer.optimize` call takes in several optional parameters that allows the caller to further fine-tune the process of optimization.
+
+```{eval-rst}
+.. autofunction:: onnxscript.optimizer.optimize
+    :noindex:
+```
+
+## Description of optimizations applied by `onnxscript.optimizer.optimize`
+
+```{eval-rst}
+.. autosummary::
+    :nosignatures:
+
+    onnxscript.optimizer.constant_folding.fold_constants
+    onnxscript.optimizer.copy_propagation.do_copy_propagation
+    onnxscript.optimizer.copy_propagation.do_sequence_simplification
```

Comment:
I think both of above (copy propagation and sequence simplification) have been directly incorporated into the constant-folding optimizer. I suspect that we may be able to eliminate these older standalone optimization components (in terms of code and files ... but those optimizations are being done, so it is okay to mention them by name, but not this specific module-name.

Reply:
I went ahead and eliminate the redundant standalone optimization components. Converted the table from autogeneration to manual mode for the time being to make it easier to highlight the passes happening inside constant_folding.